src/librustdoc/html/static/js/search.js

```diff
@@ -1066,6 +1293,70 @@ window.initSearch = function(rawSearchIndex) {
         return "<button>" + text + " <div class=\"count\">(" + nbElems + ")</div></button>";
     }
 
+    // This whole function display the query as seen by the query engine. It's helpful for users
+    // who want to see how it works a bit more in details.
+    function showQueryInfo(query) {
+        function showItemInfo(item, title) {
+            var s = "<code style='white-space:pre;'>";
+            if (title) {
+                s += title + ": ";
+            }
+            s += "{<div style=\"padding-left:25px;\">";
+            var extraBackline = "";
+
+            if (item.name) {
+                s +=  "name: <b>" + escape(item.name) + "</b>";
+                extraBackline = "<br>";
+            }
+            if (item.path) {
+                s += extraBackline + "path: <b>" + escape(JSON.stringify(item.path)) + "</b>";
+                extraBackline = "<br>";
+            }
+            if (item.isExact) {
+                s += extraBackline + "exact_search: <b>true</b>";
+                extraBackline = "<br>";
+            }
+            if (item.generics && item.generics.length !== 0) {
+                var extra = [];
+                showItemsInfo(item.generics, "generics", extra);
+                s += extraBackline + extra.join("<br>");
+            }
+            return s + "</div>}</code>";
+        }
+        function showItemsInfo(items, title, extra) {
+            if (items.length === 0) {
+                return;
+            }
+            var s = title + ": [<br><div style=\"padding-left:25px;\">";
+            for (var i = 0, len = items.length; i < len; ++i) {
+                if (i > 0) {
+                    s += "<br>";
+                }
+                s += showItemInfo(items[i]);
+            }
+            s += "</div>]";
+            extra.push(s);
+        }
+
+        if (query.foundElems === 0) {
+            return "";
+        }
+        var extra = [];
+
+        if (query.typeFilter !== NO_TYPE_FILTER) {
+            extra.push("type: <b>" + itemTypes[query.typeFilter] + "</b>");
+        }
+        if (query.elemName) {
+            extra.push(showItemInfo(query.elemName, "item info"));
+        }
+        showItemsInfo(query.elems, "unspecified elems", extra);
+        showItemsInfo(query.args, "arguments", extra);
+        showItemsInfo(query.returned, "returned", extra);
+        return "<details><summary>Parameters of the query</summary><pre>" +
+            extra.join("<br>") +
+            "</pre></details>";
+    }
```

Comment:
As an alternative to removing this entirely, maybe it can be set up so that it is gated behind a magic word in the search box? For example, while `Foo` would just return the results, `EXPLAIN: Foo` (all caps required, so that nobody uses it by accident) would display this (sans details wrapping) at the top?

Reply:
That could work! What do you think @camelid ?

Reply:
Hmm, I'm still wary of exposing this to users. What's the reason to keep it? Is it to make debugging easier?

Reply:
It was helpful for me when I was testing it. But it seems like the general opinion is that we'd better not use it, so I'll remove it.